pkgs/unified_analytics/lib/src/initializer.dart

```diff
@@ -60,7 +60,16 @@ class Initializer {
     required int toolsMessageVersion,
   }) {
     configFile.createSync(recursive: true);
-    configFile.writeAsStringSync(kConfigString);
+
+    // If the user was previously opted out, then we will
+    // replace the line that assumes automatic opt in with
+    // an opt out from the start
+    if (legacyOptOut(fs: fs, home: homeDirectory)) {
```

Comment:
Unless I'm reading this package's code wrong, it looks like this will be `platform.environment['UserProfile']` on windows (https://github.com/dart-lang/tools/blob/main/pkgs/unified_analytics/lib/src/utils.dart#L78), when the tool will be writing to `platform.environment['AppData']` on Windows https://github.com/dart-lang/usage/blob/master/lib/src/usage_impl_io.dart#L89

Reply:
Oh that's a very good catch. In the PDD, it says all files will be written to the user's home directory, am I correct to assume the home directory on windows is stored in `UserProfile`?

Reply:
my understanding is that "home directory" is an imprecise term, and especially on Windows there is not one "correct" answer for where a user's home directory is across different Windows versions and also application usages. In other words:

1. for the purposes of this PR, I think we need to check for legacy opt out from where package:usage is writing it, which is the AppData env var if present (not sure if it's ever not present, or what package:usage does in that case).
2. for the purposes of fulfilling the PDD, I'm not sure what we should do.

Reply:
The home directory as the value of the UserProfile environment variable seems correct to me and fulfills the PDD.  The most common definitions of "home directory" under Windows align with this.

Reply:
We probably should adjust the PDD to allow writing the files in a more Windows-friendly location like AppData.

Reply:
I think we would be okay if we changed the env variable for home on windows to point to `APPDATA` then? That way we can continue what was done in the past and address @christopherfujino 's item #1 above?